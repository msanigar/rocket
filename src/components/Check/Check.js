import React, { Component } from "react";

export default class Check extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isChecked: true
    };
    this.toggle = this.toggleChange.bind(this);
  }

  toggleChange() {
    this.setState({
      isChecked: !this.state.isChecked
    });
    console.log('changed!')
  }

  render() {
    return (
      <label>
        <input
          type="checkbox"
          checked={this.state.isChecked}
          onChange={this.toggle}
        />
        Check Me!
      </label>
    );
  }
}
