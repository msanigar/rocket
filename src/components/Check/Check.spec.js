import React from 'React';
import Check from './Check';
import renderer from 'react-test-renderer';

it('Check should match snapshot', () => {
    const check = renderer
        .create(
            <Check />)
        .toJSON();
    expect(check).toMatchSnapshot();
});
