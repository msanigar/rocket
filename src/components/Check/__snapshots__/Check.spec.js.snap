// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check should match snapshot 1`] = `
<label>
  <input
    checked={true}
    onChange={[Function]}
    type="checkbox"
  />
  Check Me!
</label>
`;
